src/internal/starlark/lib/k8s/doc.star

```diff
@@ -0,0 +1,99 @@
+"""
+The k8s module allows access to the Kubernetes API.
+
+In personalities where Kubernetes is available, a k8s object will be available to code.  The k8s
+object contains an attribute for each resource discovered on the server, like "pods", "services",
+etc.  Each of these attributes contains a client for accessing that resource.  No access outside the
+configured namespace is available, except for namespace-less resources (like "nodes").
+
+The k8s module is built by talking to the server to see what resources it has.  You must test for
+existence of a resource before using it:
+
+    if "pods" in dir(k8s):
+        k8s.pods.list()
+
```

Comment:
Can one check if `k8s.pods == None` or something similar?

Reply:
Nope, there is no hard-coded list of things that k8s can do anywhere in the dynamic client (or in the GVK format anywhere).  All that code is in the APIserver, and the static client is just a snapshot of a random subset of APIs that happen to be around when it's generated.  Specifically, people will use this to get things like cert-manager or istio resources; if you do the `dir` check, then you can access those if the cluster has them.